tests/unit_tests/ode_solver_unit_test/explicit_ode_solver.cpp

```diff
@@ -0,0 +1,217 @@
+
+        //initial_time_step is not modified by explicit ODE solver
+        all_parameters.ode_solver_param.initial_time_step = dt;
+        std::cout << "Using time step = " << dt << std::endl;
+        std::cout << "refinement = " << refinement << std::endl;
+
+        all_parameters.ode_solver_param.output_solution_every_x_steps = int(finalTime/dt/10.0); //output 10 vtk files (if dt reaches finalTime exactly)
+
+        unsigned int space_poly_degree = rk_order + 2;
+        std::shared_ptr < PHiLiP::DGBase<dim, double> > dg = PHiLiP::DGFactory<dim,double>::create_discontinuous_galerkin(&all_parameters, space_poly_degree, grid);
+        dg->allocate_system ();
+
+        //initial conditions
+        std::cout << "Implement initial conditions" << std::endl;
+        dealii::FunctionParser<dim> initial_condition;
+        std::string variables;
+        std::map<std::string,double> constants;
+        constants["pi"] = dealii::numbers::PI;
+        std::string expression_initial;
+        if (dim == 2){
+            variables = "x,y";
+            expression_initial = "exp( -( 20*(x-1)*(x-1) + 20*(y-1)*(y-1) ) )";
+        }
+        else if (dim == 1){
+            variables = "x";
+            //expression_initial = "exp(- 20 * (x-1) * (x-1))";	
+            expression_initial = "sin(2*pi*x/2.0)";	
+        }
+        initial_condition.initialize(variables,
+                expression_initial,
+                constants);
+        dealii::VectorTools::interpolate(dg->dof_handler,initial_condition,dg->solution);
+
+        // Create ODE solver using the factory and providing the DG object
+        std::shared_ptr<PHiLiP::ODE::ODESolverBase<dim, double>> ode_solver = PHiLiP::ODE::ODESolverFactory<dim, double>::create_ODESolver(dg);
+
+        //Solve
+        ode_solver->advance_solution_time(finalTime);
+
+        //Defining exact solution
+        dealii::FunctionParser<dim> exact_solution;
+        constants["a_x"] = adv_speed_x;
+        constants["a_y"] = adv_speed_y;
+        constants["t"] = ode_solver -> current_time;//finalTime;
+        std::string expression_exact;
+        if (dim == 2){
+            expression_exact = "exp( -( 20*(x-1-a_x*t)*(x-1-a_x*t) + 20*(y-1-a_y*t)*(y-1-a_y*t) ) )";
+        }
+        else if (dim == 1){
+            expression_exact = "sin(2*pi*(x-a_x*t)/2.0)";
+            //expression_exact = "exp( - 20*(x-1-a_x*t)*(x-1-a_x*t)) ";
+        }
+        exact_solution.initialize(variables,
+                expression_exact,
+                constants);
+
+        //Calculating L2 error
+        dealii::Vector<double> difference_per_cell(grid->n_active_cells());
+        dealii::VectorTools::integrate_difference(dg->dof_handler, 
+                dg->solution, 
+                exact_solution, 
+                difference_per_cell, 
+                dealii::QGauss<dim>(space_poly_degree+1), //check that this is correct polynomial degree
```

Comment:
Maybe you want to overintegrate? With a p+1 integration, you might have some error in your error calculation.

I would try out something like p+5 or p+10 just to see if you get better/worse convergence rates.

Reply:
There is only a small change when overintegrating, e.g. 1.2667e-10 with p+1 to 1.2890e-10 with p+10, I think because the sine wave initial condition yields a smooth enough solution. I will increase the order as you've recommended for a more robust error calculation if another user uses a less "nice" initial condition. Thanks!